### PR TITLE
Implement basic login

### PR DESCRIPTION
**Pull request by lumosnysm**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -10,6 +10,8 @@
 // Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
 // about supported directives.
 //
+//= require jquery
```
Comment (huuphuc93) [inline app/assets/javascripts/application.js]: remove all comments before this line

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  def new
+  end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
+      log_in user
+      redirect_to user
+    else
+      flash[:danger] = t(".flash_message")
```
Comment (huuphuc93) [inline app/controllers/sessions_controller.rb]: t(".flash_message") => t ".flash_mesage"

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def log_in(user)
```
Comment (huuphuc93) [inline app/helpers/sessions_helper.rb]: remove ()

File: app/views/layouts/_header.html.erb [hunk 4]
```diff
@@ -3,10 +3,27 @@
     <%= link_to "sample app", root_path, id: "logo" %>
```
Comment (huuphuc93) [inline app/views/layouts/_header.html.erb]: I18n

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -3,10 +3,27 @@
...
-        <li><%= link_to "Home", root_path %></li>
-        <li><%= link_to "Help", help_path %></li>
-        <li><%= link_to "Log in", '#' %></li>
+        <li><%= link_to t(".home"), root_path %></li>
+        <li><%= link_to t(".help"), help_path %></li>
+        <% if logged_in? %>
+          <li><%= link_to t(".user"), '#' %></li>
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              Account <b class="caret"></b>
```
Comment (huuphuc93) [inline app/views/layouts/_header.html.erb]: I18n

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -3,10 +3,27 @@
     <%= link_to "sample app", root_path, id: "logo" %>
     <nav>
       <ul class="nav navbar-nav navbar-right">
-        <li><%= link_to "Home", root_path %></li>
-        <li><%= link_to "Help", help_path %></li>
-        <li><%= link_to "Log in", '#' %></li>
+        <li><%= link_to t(".home"), root_path %></li>
+        <li><%= link_to t(".help"), help_path %></li>
+        <% if logged_in? %>
+          <li><%= link_to t(".user"), '#' %></li>
```
Comment (huuphuc93) [inline app/views/layouts/_header.html.erb]: use " instead of '

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -3,10 +3,27 @@
...
+          <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              Account <b class="caret"></b>
+            </a>
+            <ul class="dropdown-menu">
+              <li><%= link_to t(".profile"), current_user %></li>
+              <li><%= link_to t(".setting"), '#' %></li>
```
Comment (huuphuc93) [inline app/views/layouts/_header.html.erb]: use " instead of '

File: app/views/shared/_error_messages.html.erb [hunk 8]
```diff
@@ -1,7 +1,8 @@
 <% if @user.errors.any? %>
   <div id="error_explanation">
     <div class="alert alert-danger">
-      The form contains <%= pluralize @user.errors.count, "error" %>.
+      <%= t(".message") %>
```
Comment (huuphuc93) [inline app/views/shared/_error_messages.html.erb]: remove ()

File: config/locales/en.yml [hunk 9]
```diff
@@ -2,5 +2,30 @@ en:
   users:
```
Comment (huuphuc93) [inline config/locales/en.yml:2]: add file vi.yml in config/locales/vi.yml, to use vietnamese in i18n

File: app/controllers/sessions_controller.rb [hunk 10]
```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
```
Comment (huuphuc93) [inline app/controllers/sessions_controller.rb]: change to def new; end

File: app/controllers/sessions_controller.rb [hunk 11]
```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by email: params[:session][:email].downcase
+    if user && user.authenticate(params[:session][:password])
```
Comment (mdmahmudurrahman) [inline app/controllers/sessions_controller.rb]: user&.authenticate params[:session][:password]

File: config/routes.rb [hunk 12]
```diff
@@ -1,10 +1,14 @@
 Rails.application.routes.draw do
+  get "sessions/new"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Do you need this? If so, then why you need this?

File: config/routes.rb [hunk 13]
```diff
@@ -1,10 +1,14 @@
...
   get "users/new"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Do you need this? If so, then why you need this?

File: config/locales/vi.yml [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+      signup: "Đăng ký ngay"
+      login: "Đăng nhập"
+      heading: "Đăng nhập"
+      message: "Người dùng mới?"
+    create:
+      flash_message: "Email/mật khẩu không hợp lệ"
+
+  layouts:
+    header:
+        brand: "Sample app"
```
Comment (mdmahmudurrahman) [inline config/locales/vi.yml]: 2 space indentation

File: config/locales/vi.yml [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+        home: "Trang chủ"
+        help: "Trợ giúp"
+        profile: "Hồ sơ"
+        setting: "Cài đặt"
+        login: "Đăng nhập"
+        logout: "Đăng xuất"
+        account: "Tài khoản"
+
```
Comment (mdmahmudurrahman) [inline config/locales/vi.yml]: remove this empty line

File: config/locales/vi.yml [hunk 16]
```diff
@@ -0,0 +1,36 @@
+vi:
+  users:
+      new:
```
Comment (mdmahmudurrahman) [inline config/locales/vi.yml]: 2 space indentation

File: config/locales/en.yml [hunk 17]
```diff
@@ -2,5 +2,35 @@ en:
...
+    header:
+        brand: "Sample app"
+        home: "Home"
+        help: "Help"
+        profile: "Profile"
+        setting: "Setting"
+        login: "Log in"
+        logout: "Log out"
+        account: "Acount"
+
```
Comment (mdmahmudurrahman) [inline config/locales/en.yml]: remove this empty line

File: config/locales/en.yml [hunk 18]
```diff
@@ -2,5 +2,35 @@ en:
...
+      signup: "Sign up now!"
+      login: "Log in"
+      heading: "Log in"
+      message: "New user?"
+    create:
+      flash_message: "Invalid email/password combination"
+
+  layouts:
+    header:
+        brand: "Sample app"
```
Comment (mdmahmudurrahman) [inline config/locales/en.yml]: 2 space indentation

File: config/locales/en.yml [hunk 19]
```diff
@@ -1,6 +1,36 @@
...
+    header:
+      brand: "Sample app"
+      home: "Home"
+      help: "Help"
+      profile: "Profile"
+      setting: "Setting"
+      login: "Log in"
+      logout: "Log out"
+      account: "Acount"
+
```
Comment (mdmahmudurrahman) [inline config/locales/en.yml]: Can you remove this empty line?

File: config/locales/vi.yml [hunk 20]
```diff
@@ -0,0 +1,36 @@
...
+    header:
+      brand: "Sample app"
+      home: "Trang chủ"
+      help: "Trợ giúp"
+      profile: "Hồ sơ"
+      setting: "Cài đặt"
+      login: "Đăng nhập"
+      logout: "Đăng xuất"
+      account: "Tài khoản"
+
```
Comment (mdmahmudurrahman) [inline config/locales/vi.yml]: Same here

File: app/views/sessions/new.html.erb [hunk 21]
```diff
@@ -0,0 +1,22 @@
+<% provide :title, t(".title") %>
+<h1><%= t ".heading" %></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for :session, url: login_path do |f| %>
+
```
Comment (huuphuc93) [inline app/views/sessions/new.html.erb]: remove all empty line in this file

File: app/views/users/new.html.erb [hunk 22]
```diff
@@ -15,10 +15,10 @@
       <%= f.label :password %>
       <%= f.password_field :password, class: "form-control" %>
```
Comment (huuphuc93) [inline app/views/users/new.html.erb]: remove all empty line in this file

**Review by huuphuc93 — APPROVED** [review]

**Pull request merged**